### ISSUE TITLE
Allow debug

### ISSUE THREAD

**Issue by pflima92** [jponge/vertx-gradle-plugin#4]
Hi, 

The usage of plugin it's very simple, nice work here.

I suggest allow debug with plugin usage, using one IDE with Gradle this option is essential to dev. =)

Thanks in advance.

**Comment by jponge** [comment]
Thanks!

You mean running with a debugger, right?

**Comment by tiaped** [comment]
I believe it is it.
It would be really useful

**Comment by tsegismont** [comment]
I believe @pflima92 is talking about something like https://vmp.fabric8.io/#vertx:debug

**Comment by pflima92** [comment]
Exactly @tsegismont for the plugin usage the experience of this abstracted is better.

Cheers.

**Comment by tiaped** [comment]
I agree.
I made some tests locally by using -jvm-debug,
Or adding  jvmArgs = ["-Xdebug", "-Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005"] in vertx config
`vertx {
    mainVerticle = 'org.egedede.plop.MainVerticle'
    config='config/config.json'
    vertxVersion='3.5.0'
    jvmArgs = ["-Xdebug", "-Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005"]
}`

I can attach the debugger on port 5005. But breakpoints do not work :)
I'm using Intellij : 
IntelliJ IDEA 2017.3 EAP (Community Edition)
Build #IC-173.3531.6, built on October 31, 2017
JRE: 1.8.0_112-b15 amd64
JVM: Java HotSpot(TM) 64-Bit Server VM by Oracle Corporation
Linux 3.13.0-135-generic

Can anyone give a shot to this solution? To see if it is just me :)

**Comment by jponge** [comment]
From https://vmp.fabric8.io/#vertx:debug

> IMPORTANT: the redeploy mode and debug cannot be used together. When vertx:debug is launched, the redeployment mode is disabled.

We are using the same mechanism here. @tiaped @pflima92 did you disable auto-redeployment?

**Comment by tiaped** [comment]
Hi @jponge ,

I have not. Thanks for the quick answer. 
But I have an issue. If I add 'redeploy=false' to  vertx config, I have following output : 
./run.sh

```
> Task :vertxRun
Listening for transport dt_socket at address: 5005
The command '--conf' is not a valid command.
See 'java io.vertx.core.Launcher --help'

BUILD SUCCESSFUL in 6s

```
Here is the vertx conf used : 
```
vertx {
    mainVerticle = 'org.egedede.twitter.MainVerticle'
    config='config/config.json'
    vertxVersion='3.5.0'
    jvmArgs = ["-Xdebug", "-Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005"]
    redeploy=false
}
```
I will investigate more in next days.

**Comment by jponge** [comment]
Oh well, I just realized the code that assembles a command-line is wrong
when not redeploying.

**Comment by jponge** [comment]
Hum it's in configureVertxRunTask from VertxPlugin.kt, I need to
investigate what's wrong here.

**Comment by jponge** [comment]
@tiaped How about https://github.com/jponge/vertx-gradle-plugin/pull/6?

**Comment by tiaped** [comment]
@jponge It seems to work for me.
I stopped at a break point :)
May be creating a vertxdebug task could be useful. Redeployment would not be available.
I'll give a try on my own in order to learn a little bit of gradle plugin development.

**Comment by jponge** [comment]
Good, I'm merging #6 then!

Yes, let's make a `vertxDebug` specific task.

**Comment by tiaped** [comment]
If you are not in a hurry I can provide it in few weeks.
If you are then enjoy yourself.

**Comment by jponge** [comment]
Fixed by #7
